x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -356,6 +372,41 @@ public IndexInput openInput(final String name, final IOContext context) throws I
         }
     }
 
+    private void registerEvictionListener(BlobStoreIndexShardSnapshot.FileInfo fileInfo) throws IOException {
+        if (recoveryStateRegisteredEvictionListeners.add(fileInfo.physicalName())) {
+            try {
+                CacheFile cacheFile = getCacheFile(createCacheKey(fileInfo.physicalName()), fileInfo.length());
```

Comment:
Registering another EvictionListener looks trappy to me as we can't really be sure that the CacheFile instance returned here will be the same as the one used in CachedBlobContainerIndexInput.readInternal().

I'm wondering if we could instead push down the logic of resetting the recovery file details into CacheFileReference ? So that every time a new CacheFileReference is acquired we reset the recovery details? WDYT?

Reply:
I tried that approach, the problem was that since we release the reference to the `CacheFile` once we close `CachedBlobContainerIndexInput`, it's possible that when we clear the cache the eviction listener on `CacheFileReference` is not triggered as the reference was released already. Alternatively what we could do is reseting the recovered bytes in `SearchableSnapshotDirectory#clearCache` and push down the eviction logic to `CacheFileReference`. WDYT?

Reply:
Yes, I think it is worth trying this